README.md

```diff
@@ -1,70 +1,104 @@
+## What's Different
 
-Also checkout [Mongoose Delete](https://github.com/dsanel/mongoose-delete) by [Sanel Deljkic](https://github.com/dsanel).  
+In the original plugin, models were deleted with a date reference only. This version takes that and uses a Boolean flag to to mark models deleted/restored. Adds `deletedAt` field to record when a document was deleted. Additionally, it removes a lot of overhead from the original code and doesn't use Coffeescript.
 
-## License  
-This plugin is licensed under the MIT license and can ve viewed in the LICENSE file.  
+Also checkout [Mongoose Delete](https://github.com/dsanel/mongoose-delete) by [Sanel Deljkic](https://github.com/dsanel).
+
+## License
+
+This plugin is licensed under the MIT license and can ve viewed in the LICENSE file.
+
+## Installation
+
+Install using [npm](https://npmjs.org)
 
-## Installation  
-Install using [npm](https://npmjs.org)  
 ```
 npm install mongoose-softdelete --save
-```  
+```
+
+## Tests
 
-## Tests  
-IMPORTANT: You need to have MongoDB running to run tests  
+IMPORTANT: You need to have MongoDB running to run tests
 
 ```
 npm test
 ```
 
-## Usage  
+## Usage
 
-**models/test.js**  
+**models/test.js**
 
 ```js
-var mongoose = require('mongoose'),
+const mongoose = require('mongoose'),
     Schema = mongoose.Schema,
     soft_delete = require('mongoose-softdelete');
 
-var TestSchema = new Schema({
-  somefield: { type: String, default: 'Hello World!'}
+const TestSchema = new Schema({
+  somefield: { type: String, default: 'Hello World!' },
 });
 
 TestSchema.plugin(soft_delete);
 
-mongoose.model('Test', TestSchema);  
-```  
+mongoose.model('Test', TestSchema);
+```
 
-**controllers/test.js**  
+**controllers/test.js**
 
-```js  
-var Test = mongoose.model('Test');
-var test = new Test();
+```js
+const Test = mongoose.model('Test');
+const test = new Test();
 
-test.softdelete(function(err, newTest) {
-  if (err) { callback(err); }  
+test.softdelete(function (err, newTest) {
+  if (err) {
+    callback(err);
+  }
   callback(null, newTest);
 });
 
-test.restore(function(err, newTest) {
-  if (err) { callback(err); }  
+test.restore(function (err, newTest) {
+  if (err) {
+    callback(err);
+  }
   callback(null, newTest);
-});  
+});
 
 // chainable query method
 // defaults to true unless specified
 Test.find().isDeleted(false).exec();
-```  
- 
+```
+
+## Typescript
+
+```ts
+import { Schema, model } from 'mongoose';
+import softdelete, { ISoftDeletedDocument } from 'mongoose-softdelete';
+
+interface ITestDocument extends ISoftDeletedDocument {
+    somefield: string;
+}
+
+const TestSchema = new Schema({
+    somefield: { type: String, default: 'Hello World!' }
+});
+
+TestSchema.plugin(softdelete);
+
+const Test = model<ITestDocument>('Test', TestSchema);
+const test1 = new Test();
+
+test1.softdelete(function (err, newTest: ITestDocument) {
```

Comment:
TS complains about this with the following message - 
`Property 'somefield' is missing in type 'ISoftDeletedDocument' but required in type 'ITestDocument'`

Reply:
I don't understand why Typescript speak about `ISoftDeletedDocument` because I created `ITestDocument`.

Also which version of Typescript do you use ? In my case I have not any errors. I create an `sample.ts` on the root of project with folowwing content

~~~ts
import softdelete from './src/mongoose-softdelete';
import { ISoftDeletedDocument } from './index';
import { Schema, model } from 'mongoose';

interface ITestDocument extends ISoftDeletedDocument {
    somefield: string;
}

const TestSchema = new Schema({
    somefield: { type: String, default: 'Hello World!' }
});

TestSchema.plugin(softdelete);

const Test = model<ITestDocument>('Test', TestSchema);
const test1 = new Test();

test1.softdelete(function (err, newTest: ITestDocument) {
    // ...
});
~~~